File: programmers/240408_lotto.py
# https://school.programmers.co.kr/learn/courses/30/lessons/77484?language=python3﻿
def solution(lottos, win_nums):
    #iterator
    i = 0
    #min and max value of winning number or ranks
    min, max = 0, 0
    #number of 0 in list lottos
    numOfZero = 0
    #rank dictionary (winning number : rank)
    rank = {6:1, 5:2, 4:3, 3:4, 2:5, 1:6, 0:6}

    #count number of winning number(min & max)
    for i in lottos:
        if (i in win_nums): # if lottos[n] is in list win_nums
            min += 1
            max += 1
        elif (i == 0): # if lottos[n] is 0 (unknown value)
            numOfZero += 1
        else:
            continue

    # 0 means that it can be a winning number
    max += numOfZero

    #change min & max values to its ranks
    min = rank[min]
    max = rank[max]

    return [max, min]

#test code
print(solution([44, 1, 0, 0, 31, 25], [31, 10, 45, 1, 6, 19]))

File: programmers/240408_lotto.cpp
#include <string>
#include <vector>
#include <iostream>
#include <algorithm>

using namespace std;

int returnRank(int num) //return ranks of lotto
{
    if (num < 2) {return 6;}       //hit nums < 2, rank 6th
    else if (num == 2) {return 5;} //hit nums = 2, rank 5th
    else if (num == 3) {return 4;} //hit nums = 3, rank 4th
    else if (num == 4) {return 3;} //hit nums = 4, rank 3rd
    else if (num == 5) {return 2;} //hit nums = 5, rank 2nd
    else if (num == 6) {return 1;} //hit nums = 6, rank 1st
    else
    {
        cout << "input number is over 6";
        return 0;
    }
};

vector<int> solution(vector<int> lottos, vector<int> win_nums)
{
    vector<int> answer;  //return value
    int numOfZero = 0;   //number of 0 in lottos
    int min = 0;    //minimum number of hit numbers
    int max = 0;    //maximum number of hit numbers

    for (int elem:lottos)
    {
      //if there's same number in win_nums(which means one number of lottos hit)
        if (find(begin(win_nums), end(win_nums), elem) != end(win_nums))
        {
          //count as hit number
            min++;
            max++;
        }
        else if (elem == 0) //count number of 0
        {
            numOfZero ++;
        }
    }

    max += numOfZero;

    answer = {returnRank(max), returnRank(min)}; //change hit numbers to its ranks

    return answer;
};

int main(void) { //test case
    vector<int> input1{44, 1, 0, 0, 31, 25};
    vector<int> input2{31, 10, 45, 1, 6, 19};
    vector<int> a = solution(input1, input2);
    for(int&i : a)
    {
        cout << i << " ";
    }
};
